python/src/tasks/list_genes.py

```diff
@@ -19,66 +23,33 @@ def _format_result(self, result, total):
         return [Result(result)]
 
     def compute(self):
-        genes = self.adata.var
-
-        # Fields to return.
-        select_fields = self.task_def["selectFields"]
-
-        # if there is no search pattern defined, do not restrict gene names
-        filter_pattern = self.task_def.get("geneNamesFilter", None)
-
-        if filter_pattern and "gene_names" in select_fields:
-            filter_query = "WHERE gene_names LIKE '{}'".format(filter_pattern)
-        else:
-            filter_query = ""
-
-        # What the fields are ordered by
-        order_by = self.task_def["orderBy"]
-
-        # Order direction ('asc' or 'desc')
-        order_direction = self.task_def["orderDirection"].upper()
-
-        # Return only from this index
-        offset = int(self.task_def["offset"])
-
-        # How many to to return.
-        limit = int(self.task_def["limit"])
-
-        # Set up SQL query and PandaSQL for efficient querying.
-        query = """
-            SELECT {}, count(*) OVER() AS full_count
-            FROM genes
-            {}
-            ORDER BY {} {}
-            LIMIT {}
-            OFFSET {}
-        """
-        execute_query = lambda q: sqldf(q, {"genes": genes})
-
-        query = query.format(
-            ", ".join(select_fields),
-            filter_query,
-            order_by,
-            order_direction,
-            limit,
-            offset,
+        request = self.task_def
+        #
+        # Remove potentialy hazardous characters from the names filter
+        # Leaving this'd leave us open to a DDOS attack in the form of a very time consuming regex.
+        # More info here: https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS
+        #
+        # The symbols ^ and $ can't be removed because they are the way the UI indicates the type of search the user is performing.
+        #
+        if "geneNamesFilter" in request:
+            geneFilter = request["geneNamesFilter"]
+            regexChars = "{}|()?¿*+|\/.<>"
+            for char in regexChars:
+                geneFilter = geneFilter.replace(char, "")
+            request["geneNamesFilter"] = geneFilter
+        r = requests.post(
+            f"{config.R_WORKER_URL}/v0/listGenes",
+            headers={"content-type": "application/json"},
+            data=json.dumps(request),
         )
-        result = execute_query(query)
-
-        # Get total number of results if there are results
+        resR = r.json()
+        # Convert to dataframe to prepare the data for the UI
+        resR = pd.DataFrame(resR)
         total = 0
-
-        if len(result) > 0:
-            total = result["full_count"][0]
-
+        if len(resR) > 0:
+            total = resR["full_count"][0]
+        resR = resR.drop("full_count", axis=1)
```

Comment:
maybe put the logic into R to have all computing at the same place (use transpose() ?)